### PR TITLE
Use k8s.io/component-base/cli to run the command

### PR DESCRIPTION
**Pull request by liangyongzhenya**
Use k8s.io/component-base/cli to run the command

### CODE REVIEW

**Comment by RuliXu** [comment]
We can link the related issue here in the PR description, could make it more clear.

**Comment by liangyongzhenya** [comment]
> We can link the related issue here in the PR description, could make it more clear.

> Use k8s.io/component-base/cli to run the command

https://github.com/clusterpedia-io/clusterpedia/issues/226

**Comment by liangyongzhenya** [comment]
> We can link the related issue here in the PR description, could make it more clear.

Thanks so much, already linked to the issue.

File: cmd/apiserver/main.go [hunk 1]
```diff
@@ -1,29 +1,25 @@
 package main
 
 import (
-	"fmt"
-	"os"
-
+	_ "fmt"
```
Comment (RuliXu) [inline cmd/apiserver/main.go:4]: The import order and format need to be ordered.

File: cmd/clustersynchro-manager/main.go [hunk 2]
```diff
@@ -1,7 +1,8 @@
 package main
 
 import (
-	"fmt"
+	"github.com/spf13/cobra"
```
Comment (RuliXu) [inline cmd/clustersynchro-manager/main.go:4]: Same here, the import order and format need to be ordered.

File: cmd/apiserver/main.go [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
 	"os"
 
-	apiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/component-base/cli"
 	"k8s.io/component-base/logs"
 
+	apiserver "k8s.io/apiserver/pkg/server"
+
 	"github.com/clusterpedia-io/clusterpedia/cmd/apiserver/app"
+	"github.com/spf13/cobra"
```
Comment (RuliXu) [inline cmd/apiserver/main.go:12]: Normally, we divide import into 3 parts:
1. Standard libraries
2. Third-party libraries 
3. Project-owned libraries, in this case, clusterpedia-owned libraries

File: cmd/clustersynchro-manager/main.go [hunk 4]
```diff
@@ -1,34 +1,31 @@
...
 
 import (
-	"fmt"
-	"math/rand"
 	"os"
 	"time"
 
+	"github.com/spf13/cobra"
+	"k8s.io/component-base/cli"
+	"math/rand"
```
Comment (RuliXu) [inline cmd/clustersynchro-manager/main.go:9]: For example, `math/rand` belongs to standard libraries.

File: cmd/apiserver/main.go [hunk 5]
```diff
@@ -1,29 +1,27 @@
...
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
+	command := runApiServerCommand()
+	code := cli.Run(command)
+	os.Exit(code)
 }
 
-func runApiServerCommand() error {
+func runApiServerCommand() *cobra.Command {
```
Comment (RuliXu) [inline cmd/apiserver/main.go:21]: As we used `k8s.io/component-base/cli`, we don't need `runApiServerCommand()` anymore.

File: cmd/clustersynchro-manager/main.go [hunk 6]
```diff
@@ -1,34 +1,31 @@
...
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
+	command := runClusterSynchroManagerCommand()
+	code := cli.Run(command)
+	os.Exit(code)
 }
 
-func runClusterSynchroManagerCommand() error {
+func runClusterSynchroManagerCommand() *cobra.Command {
```
Comment (RuliXu) [inline cmd/clustersynchro-manager/main.go:23]: As we used `k8s.io/component-base/cli` , we don't need `runClusterSynchroManagerCommand()` anymore.

Comment (RuliXu) [inline cmd/clustersynchro-manager/main.go:23]: As we used `k8s.io/component-base/cli` , we don't need `runClusterSynchroManagerCommand()` anymore.

File: cmd/apiserver/main.go [hunk 7]
```diff
@@ -1,29 +1,27 @@
...
 	logs.InitLogs()
```
Comment (RuliXu) [inline cmd/apiserver/main.go:22]: As we used `k8s.io/component-base/cli` , we don't need to init log anymore.

Comment (RuliXu) [inline cmd/apiserver/main.go:22]: As we used `k8s.io/component-base/cli` , we don't need to init log here.

File: cmd/clustersynchro-manager/main.go [hunk 8]
```diff
@@ -1,34 +1,31 @@
...
 	rand.Seed(time.Now().UnixNano())
 
 	logs.InitLogs()
```
Comment (RuliXu) [inline cmd/clustersynchro-manager/main.go:26]: As we used `k8s.io/component-base/cli` , we don't need to init log anymore.

Comment (RuliXu) [inline cmd/clustersynchro-manager/main.go:26]: As we used `k8s.io/component-base/cli` , we don't need to init log anymore.

File: cmd/clustersynchro-manager/main.go [hunk 9]
```diff
@@ -1,34 +1,18 @@
 package main
 
 import (
-	"fmt"
-	"math/rand"
 	"os"
-	"time"
+
+	"k8s.io/component-base/cli"
```
Comment (RuliXu) [inline cmd/clustersynchro-manager/main.go:7]: Unnecessary blank.

**Comment by RuliXu** [comment]
LGTM

**Comment by RuliXu** [comment]
@Iceber /cc

File: cmd/apiserver/main.go [hunk 10]
```diff
@@ -1,29 +1,17 @@
 package main
 
 import (
-	"fmt"
 	"os"
 
 	apiserver "k8s.io/apiserver/pkg/server"
-	"k8s.io/component-base/logs"
+	"k8s.io/component-base/cli"
```
Comment (Iceber) [inline cmd/apiserver/main.go:7]:
```go
_ "k8s.io/component-base/logs/json/register"          // for JSON log format registration
```
we can register json log format

File: cmd/clustersynchro-manager/main.go [hunk 11]
```diff
@@ -1,34 +1,17 @@
 package main
 
 import (
-	"fmt"
-	"math/rand"
 	"os"
-	"time"
 
+	"k8s.io/component-base/cli"
```
Comment (Iceber) [inline cmd/clustersynchro-manager/main.go:6]: ditto

**Review by Iceber — COMMENTED** [review]
other looks good

File: cmd/clustersynchro-manager/main.go [hunk 12]
```diff
@@ -1,34 +1,18 @@
...
 	apiserver "k8s.io/apiserver/pkg/server"
-	"k8s.io/component-base/logs"
+	_ "k8s.io/component-base/logs/json/register"
```
Comment (Iceber) [inline cmd/clustersynchro-manager/main.go:8]: The order of package import should be:
```go
	apiserver "k8s.io/apiserver/pkg/server"
	"k8s.io/component-base/cli"
	_ "k8s.io/component-base/logs/json/register"
```

**Review by Iceber — APPROVED** [review]
LGTM, @liangyongzhenya Thanks for your contribution, and thanks to @RuliXu for the review

**Pull request merged**